libtransmission/torrent.cc

```diff
@@ -563,7 +563,12 @@ static void onTrackerResponse(tr_torrent* tor, tr_tracker_event const* event, vo
         break;
 
     case tr_tracker_event::Type::Warning:
-        tr_logAddWarnTor(tor, fmt::format(_("Tracker warning: '{warning}'"), fmt::arg("warning", event->text)));
+        tr_logAddWarnTor(
+            tor,
+            fmt::format(
+                _("Tracker warning: '{warning}' ({url})"),
+                fmt::arg("warning", event->text),
+                fmt::arg("url", event->announce_url)));
```

Comment:
This can leak PII if the user has a personal announce id embedded in that URL. Some other announcer code has this workaround:

```
        // don't log the full URL, since that might have a personal announce id
        // (note: we know 'parsed' will be successful since this url has a scrape_info)
        if (auto const parsed = tr_urlParse(url); parsed)
        {
            tr_logAddDebug(
                fmt::format(FMT_STRING("Reducing multiscrape max to {:d}"), n),
                fmt::format(FMT_STRING("{:s}://{:s}:{:d}"), parsed->scheme, parsed->host, parsed->port));
        }
```

But rather than coding that logic twice let's make a helper function in `web-utils.h` like

```c++
// Convenience function to get a log-safe version of a tracker URL.
// This is to avoid logging sensitive info, e.g. a personal announcer id in the URL
std::string tr_urlTrackerLogName(std::string_view url)
{
    if (auto const parsed = tr_urlParse(url); parsed)
    {
        return fmt::format(FMT_STRING("{:s}://{:s}:{:d}"), parsed->scheme, parsed->host, parsed->port));
    }

    // ...we have an invalid URL?
    return url;
}
```

And then use it in both places. That way the log calls wouldn't need any conditionals, just something like

```c++
            fmt::format(
                _("Tracker warning: '{warning}' ({url})"),
                fmt::arg("warning", event->text),
                fmt::arg("url", tr_urlTrackerLogName(url))));
```

*edit:* looks like announcer.cc gets this wrong in at least two places in the "Unsupported URL" messages. These should be

```diff
---             tr_logAddError(fmt::format(_("Unsupported URL: '{url}'"), fmt::arg("url", scrape_sv)));
+++             tr_logAddError(fmt::format(_("Unsupported URL: '{url}'"), fmt::arg("url", tr_urlTrackerLogName(scrape_sv))));
```

Reply:
Done. Although, I'll not change the cases of "Unsupported URL": it means it's not a valid URL and the user will want the full visibility of what is wrong with that URL.